004-directives/src/app/states/home/home.controller.js

```diff
@@ -6,16 +6,60 @@
     .controller('HomeController', HomeController);
 
   /** @ngInject */
-  function HomeController(RESERVED_BY) {
+  function HomeController($scope, RESERVED_BY) {
     var vm = this;
     vm.RESERVED_BY = RESERVED_BY;
     vm.icons = ['account_circle', 'home', 'alarm'];
     vm.card = {};
+    vm.favoriteList = [];
     vm.setIcon = setIcon;
+    vm.setFavorite= setFavorite;
+    vm.showFavorite= showFavorite;
+    vm.indexCard= undefined;
+    vm.clearForm= clearForm;
+    vm.editCard= editCard;
+    vm.removeCard= removeCard;
 
     function setIcon(iconClass) {
       vm.card.icon = iconClass;
     }
+
+    function setFavorite(icon, title, description, textColor, background) {
+      var card = {
+        icon: icon,
+        title: title,
+        description: description,
+        textColor: textColor,
+        background: background
+      };
+      vm.favoriteList.push(card);
+      vm.card = {};
+      $scope.cardForm.$setPristine();
+      vm.indexCard= undefined;
+    }
+
+    function showFavorite(card, index) {
+      vm.savedCard= true;
+      vm.card = angular.copy(card);
+      vm.indexCard = index;
+    }
+
+    function removeCard(index) {
+      vm.favoriteList.splice(index,1);
+    }
+
+    function editCard() {
+      if (vm.indexCard !== undefined) {
```

Comment:
falsy value you can use !vm.incexCard 
Although this works and is more specific but be sure what falsy values are